src/txmempool.h

```diff
@@ -138,6 +363,29 @@ class CTxMemPool
     void ApplyDeltas(const uint256 hash, double &dPriorityDelta, CAmount &nFeeDelta);
     void ClearPrioritisation(const uint256 hash);
 
+public:
+    void RemoveStaged(std::set<uint256>& stage);
+
+    /** When adding transactions from a disconnected block back to the mempool,
+     *  new mempool entries may have children in the mempool (which is generally
+     *  not the case when otherwise adding transactions).
+     *  UpdateTransactionsFromBlock will find child transactions and update the
+     *  descendant state for each transaction in hashesToUpdate (excluding any
+     *  child transactions present in hashesToUpdate, which are already accounted
+     *  for).
+     */
```

Comment:
This comment doesn't make it clear whether or not transactions in hashesToUpdate are or are not already in the mempool. :)